### DISCUSSION TITLE
Highlight menu items as you tab through completion options

### DISCUSSION THREAD

**Discussion by ldelossa** [marlonrichert/zsh-autocomplete#712] [Q&A]
Hello,

I searched the QAs for awhile but I didn't find an answer to my question. 

I use 'Tab' to cycle through completion options.

However, its a bit confusing to NOT see the currently selected item be highlighted in the menu below it. 

Is it possible to keep the menu highlight in sync with the command line option being cycled through? Or must you use the menu and the command line options out of sync?

**Comment by marlonrichert** [comment]
What you want is documented in the following two Readme sections:
* [Make <kbd>Tab</kbd> and <kbd>Shift</kbd><kbd>Tab</kbd> go to the menu](https://github.com/marlonrichert/zsh-autocomplete/blob/main/README.md#make-tab-and-shifttab-go-to-the-menu)
* [Make <kbd>Tab</kbd> and <kbd>Shift</kbd><kbd>Tab</kbd> change the selection in the menu](https://github.com/marlonrichert/zsh-autocomplete/blob/main/README.md#make-tab-and-shifttab-change-the-selection-in-the-menu)

**Comment by alexwforsythe** [comment]
I got confused by this, too, and had to read the zsh docs for a bit to figure it out. Apparently, some of the zsh completion widgets, like `complete-word` and `incremental-complete-word` show a "non-interactive" menu that can only be cycled on the command line (no highlighting). The `menu-select` widget enters an "interactive" completion menu, though, which does highlight matches, e.g.:

```sh
# Tab and shift+tab enter the interactive completion menu.
bindkey '^I' menu-select
bindkey "$terminfo[kcbt]" menu-select
```

The match colors can be customized using the `list-colors` zstyle. You'll need to read up on `LS_COLORS` formatting to understand it, though:
```sh
# my config with base16 color themes:

if ! zstyle -a ":completion:*$zs" list-colors _zs_ls_colors >/dev/null 2>&1; then
  _zs_ls_colors=()
fi
# Override the match (ma) color to be more subtle (fg lighter white, bg darkest
# gray, bold):
_zs_ls_colors+='ma=38;5;21;48;5;18;1'
zstyle ":completion:*$zs" list-colors $_zs_ls_colors
unset _zs_ls_colors
```
